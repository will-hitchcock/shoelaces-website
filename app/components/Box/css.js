import colors from 'utils/colors'

export const css = ({ color }) => ({
  padding: '1em',
  backgroundColor: color || colors.secondary
})

export default css
