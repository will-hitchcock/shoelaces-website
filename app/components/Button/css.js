import colors from 'utils/colors'

export const css = {
  display: 'inline-block',
  borderRadius: '3px',
  padding: '0.5rem 0',
  margin: '0.5rem 1rem',
  width: '11rem',
  background: colors.secondary,
  color: 'white',
  border: `2px solid ${colors.secondaryLighter}`,
  textAlign: 'center',
  textDecoration: 'none'
}

export default css
