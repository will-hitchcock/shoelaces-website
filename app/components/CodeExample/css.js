import colors from 'utils/colors'

const editor = {
  overflow: 'scroll'
}

const error = {
  borderLeft: `5px ${colors.darkRed} solid`,
  backgroundColor: colors.red,
  padding: '0.5em',
  color: colors.darkRed,
  marginTop: '0.5em'
}

export { editor, error }
