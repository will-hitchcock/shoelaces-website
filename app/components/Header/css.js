// import colors from 'utils/colors'

const shared = {
  textAlign: 'center',
  color: 'white'
}

const h1 = { ...shared }
const h2 = { ...shared }
const h3 = { ...shared }
const h4 = { ...shared }
const h5 = { ...shared }

export default { h1, h2, h3, h4, h5 }
