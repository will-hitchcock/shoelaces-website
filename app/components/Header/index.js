/**
*
* Header
*
*/

import React from 'react'
import { oneOf, node } from 'prop-types'
import styled from 'styled-components'
import css from './css'

const H1 = styled.h1`${css.h1}`
const H2 = styled.h2`${css.h2}`
const H3 = styled.h3`${css.h3}`
const H4 = styled.h4`${css.h4}`
const H5 = styled.h5`${css.h5}`

const Header = ({ type, children }) => {
  switch (type) {
    case 'h5':
      return <H5>{children}</H5>
    case 'h3':
      return <H3>{children}</H3>
    case 'h4':
      return <H4>{children}</H4>
    case 'h2':
      return <H2>{children}</H2>
    default:
      return <H1>{children}</H1>
  }
}

Header.propTypes = {
  type: oneOf(['h1', 'h2', 'h3', 'h4', 'h5']),
  children: node
}

Header.defaultProps = {
  type: 'h1'
}

export default Header
