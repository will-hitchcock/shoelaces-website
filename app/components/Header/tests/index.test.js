import React from 'react'
import Header from '../index'
import css from '../css'

describe('Header', () => {
  it('renders a <H5 />', () => {
    expect(shallow(<Header type="h5" />)).toMatchSnapshot()
  })

  it('renders a <H4 />', () => {
    expect(shallow(<Header type="h4" />)).toMatchSnapshot()
  })

  it('renders a <H3 />', () => {
    expect(shallow(<Header type="h3" />)).toMatchSnapshot()
  })

  it('renders a <H2 />', () => {
    expect(shallow(<Header type="h2" />)).toMatchSnapshot()
  })

  it('renders a <H1 />', () => {
    expect(shallow(<Header type="h1" />)).toMatchSnapshot()
  })

  it('renders a <H1 />', () => {
    expect(shallow(<Header />)).toMatchSnapshot()
  })

  it('renders css', () => {
    expect(css).toMatchSnapshot()
  })
})
