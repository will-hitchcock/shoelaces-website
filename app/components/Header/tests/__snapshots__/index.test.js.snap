exports[`Header renders a <H1 /> 1`] = `<styled.h1 />`;

exports[`Header renders a <H1 /> 2`] = `<styled.h1 />`;

exports[`Header renders a <H2 /> 1`] = `<styled.h2 />`;

exports[`Header renders a <H3 /> 1`] = `<styled.h3 />`;

exports[`Header renders a <H4 /> 1`] = `<styled.h4 />`;

exports[`Header renders a <H5 /> 1`] = `<styled.h5 />`;

exports[`Header renders css 1`] = `
Object {
  "h1": Object {
    "color": "white",
    "textAlign": "center",
  },
  "h2": Object {
    "color": "white",
    "textAlign": "center",
  },
  "h3": Object {
    "color": "white",
    "textAlign": "center",
  },
  "h4": Object {
    "color": "white",
    "textAlign": "center",
  },
  "h5": Object {
    "color": "white",
    "textAlign": "center",
  },
}
`;
